vllm/engine/arg_utils.py

```diff
@@ -409,9 +415,16 @@ def add_cli_args(
     @classmethod
     def from_cli_args(cls, args: argparse.Namespace) -> 'EngineArgs':
         # Get the list of attributes of this dataclass.
-        attrs = [attr.name for attr in dataclasses.fields(cls)]
+        attrs = [
+            attr.name for attr in dataclasses.fields(cls)
+            if attr.name != "tensorizer_args"
+        ]
         # Set the attributes from the parsed arguments.
         engine_args = cls(**{attr: getattr(args, attr) for attr in attrs})
+
+        engine_args.tensorizer_args = TensorizerArgs.from_cli_args(args) if (
+            engine_args.load_format == "tensorizer") else None
+
```

Comment:
this is pretty hacky, maybe it would be a better idea to move this logic to `create_engine_config`

Reply:
This is a bit tricky as I implemented this in a way where `tensorizer_args` is expected to be an `attr` of `EngineArgs` but not an arg for its argparser, which didn't feel great but seemed to be a necessary evil outside of creating an entirely new `TensorizerConfig`. I could propose a different solution that removes `tensorizer_args` from `EngineArgs` but is also slightly hacky.

In the `__post_init__` for `EngineArgs`..
```
    def __post_init__(self):
        if self.tokenizer is None:
            self.tokenizer = self.model
+      if not hasattr(self, 'tensorizer_args'):
+          self.tensorizer_args = None
```

With a perhaps slightly nicer `EngineArgs.from_cli_args`:

```
    @classmethod
    def from_cli_args(cls, args: argparse.Namespace) -> 'EngineArgs':
        # Get the list of attributes of this dataclass.
        attrs = [attr.name for attr in dataclasses.fields(cls)]
        # Set the attributes from the parsed arguments.
        engine_args = cls(**{attr: getattr(args, attr) for attr in attrs})

        # Check if tensorizer_args are expected.
        if args.load_format == "tensorizer":
            tensorizer_attrs = [
                attr.name for attr in dataclasses.fields(TensorizerArgs)
            ]
            # Create an instance of TensorizerArgs with its subset of the CLI args.
            engine_args.tensorizer_args = TensorizerArgs(
                **{attr: getattr(args, attr) for attr in tensorizer_attrs}
            )

        return engine_args
```

And a hacky tweak to `LLM`'s `__init__`:

```
        if "disable_log_stats" not in kwargs:
            kwargs["disable_log_stats"] = True
+      if "tensorizer_args" in kwargs:
+          tensorizer_args = kwargs["tensorizer_args"]
+          del kwargs["tensorizer_args"]
+      else:
+          tensorizer_args = None
        engine_args = EngineArgs(
            model=model,
            tokenizer=tokenizer,
            tokenizer_mode=tokenizer_mode,
            trust_remote_code=trust_remote_code,
            tensor_parallel_size=tensor_parallel_size,
            dtype=dtype,
            quantization=quantization,
            revision=revision,
            tokenizer_revision=tokenizer_revision,
            seed=seed,
            gpu_memory_utilization=gpu_memory_utilization,
            swap_space=swap_space,
            enforce_eager=enforce_eager,
            max_context_len_to_capture=max_context_len_to_capture,
            disable_custom_all_reduce=disable_custom_all_reduce,
            **kwargs,
        )
+      engine_args.tensorizer_args = tensorizer_args
        self.llm_engine = LLMEngine.from_engine_args(
            engine_args, usage_context=UsageContext.LLM_CLASS)
        self.request_counter = Counter()
```

Reply:
Let me think about this!